public/src/utils.js

```diff
@@ -14,15 +14,16 @@
 		};
 
 		process.profile = function (operation, start) {
-			console.log('%s took %d milliseconds', operation, process.elapsedTimeSince(start));
+			winston.log('%s took %d milliseconds', operation, process.elapsedTimeSince(start));
 		};
 
 		process.elapsedTimeSince = function (start) {
 			var diff = process.hrtime(start);
 			return (diff[0] * 1e3) + (diff[1] / 1e6);
 		};
 	} else {
-		window.utils = factory(window.XRegExp);
+		winston = console;
+		window.utils = factory(window.XRegExp, winston);
 	}
 }(function (XRegExp) {
```

Comment:
Change this to `(XRegExp, console)`, info changes on lines 4-6, change line 8 to `factory(require('xregexp'), winston);`, change line 26 to `factory(window.XRegExp, console);` then use `console` instead of `winston` everywhere below.

Reply:
well the point of `winston = console` is to be able to use `winston`, when available, when executing on the server. We don't need to pass `console` as an argument, it's global, on both ends.

Reply:
Right but you want to use console as a backup whenever winston is not available. Whether you want to use `console` or `winston` as the variable name doesn't really matter to me, but either way it should be done by passing them to the factory function.

I suggested using `console` because then people won't try to use winston-only methods client-side, and it will automatically pipe all console commands through winston on the server side

Reply:
Ah ok ok got it. I wanted to use `winston` as a variable name to be more consistent with the server side code, and "most" of the winston functions __we normally use__ are covered by console, but not all, so you're right, better be safe here.

Anyways, this is moot now because as of now, `utils` does not log anything, i removed the `callable()` that used spit out some warnings.

so I either remove all of these top changes or use `eslint-disable-next-line` to supress the  `no-unused-vars` lint error, see [commit](https://github.com/NodeBB/NodeBB/pull/7673/commits/335a80178be0f5242d9e892f887eaa486112d6c1#diff-89ce74219f35abaa6f40c1eb38043810R26)